src/main/java/com/xpanxion/java/springboot/da1/demo/service/student9/MemberHistoryService9.java

```diff
@@ -73,35 +85,27 @@ public StringBuilder getMemberWorkOutHistory(int id){
        return newList;
     }
 
-    public StringBuilder getMemberWorkOutLength(int id, String type){
-        var stringBuilder = new StringBuilder();
+    public WorkoutLength9 getMemberWorkOutLength(int id, String type){
         if(type.equalsIgnoreCase("min")){
             var tempList = memberHistoryRepository9.getWorkoutLengthMin(id);
-            for(var temp : tempList){
-                var duration = Duration.between(temp.getCheckIn(), temp.getCheckOut());
-                stringBuilder.append("{");
-                stringBuilder.append("\n");
-                stringBuilder.append("LengthInMinutes: ").append(duration.toMinutes());
-                stringBuilder.append("\n");
-                stringBuilder.append("Member Id: ").append(temp.getMemberId().getMemberId());
-                stringBuilder.append("\n");
-                stringBuilder.append("date: ").append(temp.getCheckIn().toLocalDate());
-            }
+            var l = tempList.stream().map(m -> {
+                var duration = Duration.between(m.getCheckIn(), m.getCheckOut());
+                var memberId=  m.getMemberId().getMemberId();
+                var date = m.getCheckIn().toLocalDate();
+                return new WorkoutLength9(duration.toMinutes(), memberId, date);
+            }).toList();
+
+            return l.get(0);
         } else if(type.equalsIgnoreCase("max")){
             var tempList = memberHistoryRepository9.getWorkoutLengthMax(id);
-            for(var temp : tempList){
-                var duration =  Duration.between(temp.getCheckIn(), temp.getCheckOut());
-                stringBuilder.append("{");
-                stringBuilder.append("\n");
-                stringBuilder.append("LengthInMinutes: ").append(duration.toMinutes());
-                stringBuilder.append("\n");
-                stringBuilder.append("Member Id: ").append(temp.getMemberId().getMemberId());
-                stringBuilder.append("\n");
-                stringBuilder.append("date: ").append(temp.getCheckIn().toLocalDate());
-                stringBuilder.append("\n").append("}");
-                stringBuilder.append("\n");
-            }
+            var l = tempList.stream().map(m -> {
+                var duration = Duration.between(m.getCheckIn(), m.getCheckOut());
+                var memberId=  m.getMemberId().getMemberId();
+                var date = m.getCheckIn().toLocalDate();
+                return new WorkoutLength9(duration.toMinutes(), memberId, date);
+            }).toList();
+            return l.get(0);
         } // else doNothing()
-        return stringBuilder;
+        return null;
```

Comment:
`return retval`;

Reply:
This is known as "single return" from functions and it might be a coding style that is enforced.

Reply:
retrieval =  l.get(0);
        } // else doNothing()
        return retrieval;